%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{0}



\title{GT\sphinxhyphen{}Plotting}
\date{24 November 2020}
\release{}
\author{Anza Ghaffar, Anibal Morales}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page 

        \begin{titlepage}

            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {GT-Plotting Documentation}}

            \vspace{0mm} %%% * is used to give space from top
            \textbf{\huge {A Step-by-Step Tutorial - V1.0}}

            \vspace{50mm}
            \textbf{\Large {Anza Ghaffar, Anibal Morales}}

            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory}}

            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria}}

	    \vspace{10mm}
            \normalsize Created on : July, 2020

            \vspace*{0mm}
            \normalsize  Last updated : November, 2020


            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}

        \end{titlepage}
        \pagenumbering{arabic}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\part{Background}
\label{\detokenize{index:background}}
This tutorial plots the genotypic similarities and differences between individuals aligned with to reference genome extracted from a \sphinxstylestrong{vcf} file. A \sphinxcode{\sphinxupquote{vcf.gz}} file is used as input, and it outputs \sphinxcode{\sphinxupquote{.png}} files, depicting chromosome number, chromosome position, and genotype.

\begin{sphinxadmonition}{note}{Note:}
This is run using Ubuntu 18.04 Bionic Beaver with R version 3.6.0. For different Ubuntu distributions, download and install the appropriate software/packages.

Code ran on Linux terminal is preceded by \sphinxcode{\sphinxupquote{\$}}, while code ran on RStudio is preceded by \sphinxcode{\sphinxupquote{\textgreater{}}}.
\end{sphinxadmonition}


\part{Installing conda and Setting Up Channels}
\label{\detokenize{index:installing-conda-and-setting-up-channels}}
It is recommended to install miniconda3 and create a conda environment to install all necessary packages and dependencies without affecting the system. Miniconda is a free minimal installer for conda. It is a small, bootstrap version of Anaconda that includes only conda, Python, the packages they depend on, and a small number of other useful packages, including pip, zlib and a few others. Use the conda install command to install 720+ additional conda packages from the Anaconda repository.

Open a new terminal \sphinxstylestrong{(Ctrl + Alt + T)} and make sure that the system is up\sphinxhyphen{}to\sphinxhyphen{}date,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt\PYGZhy{}get update
\end{sphinxVerbatim}

Download the latest package from the ‘\sphinxhref{https://docs.conda.io/en/latest/miniconda.html}{Miniconda Webpage} and install it with,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} curl \PYGZhy{}O https://repo.anaconda.com/miniconda/Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\PYGZdl{} sh Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\end{sphinxVerbatim}

It may be possible that Python 2.7 was installed with miniconda3. In order to use a newer version of Python, install the preferred Python version and update conda to resolve any dependency failures,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install \PYGZhy{}c anaconda python=3.7
\PYGZdl{} conda update \PYGZhy{}\PYGZhy{}all
\end{sphinxVerbatim}

Setup the appropriate bioconda channels. Make sure to run the following commands exactly in this order,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda config \PYGZhy{}\PYGZhy{}add channels defaults
\PYGZdl{} conda config \PYGZhy{}\PYGZhy{}add channels bioconda
\PYGZdl{} conda config \PYGZhy{}\PYGZhy{}add channels conda\PYGZhy{}forge
\end{sphinxVerbatim}

Bioconda is now enabled, so you can install any packages and versions available on the bioconda channel, such as

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install bwa bowtie bcftools=1.9
\end{sphinxVerbatim}


\part{Creating a conda environment}
\label{\detokenize{index:creating-a-conda-environment}}
Create a new conda environment with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda create \PYGZhy{}\PYGZhy{}name \PYGZlt{}environment\PYGZhy{}name\PYGZgt{}
\end{sphinxVerbatim}

Substitute \sphinxcode{\sphinxupquote{\textless{}environment\sphinxhyphen{}name\textgreater{}}} with the preferred name you want. As an example,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda create \PYGZhy{}\PYGZhy{}name R\PYGZhy{}Data\PYGZhy{}Science
\end{sphinxVerbatim}

Once the newly created environment has been installed, activate it with,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate R\PYGZhy{}Data\PYGZhy{}Science
\end{sphinxVerbatim}


\part{Installing R and RStudio}
\label{\detokenize{index:installing-r-and-rstudio}}
In order to install R and RStudio, it is recommended to run the installation with the use of conda. To install R, input the following,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install rstudio
\end{sphinxVerbatim}

By directly installing RStudio, conda will install all the necessary dependencies, including the appropriate version of R. Once the installation is complete, open RStudio in the terminal,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} rstudio
\end{sphinxVerbatim}

RStudio will open a new window where R scripts can be run.


\part{Installing and Loading the Necessary R Packages}
\label{\detokenize{index:installing-and-loading-the-necessary-r-packages}}
In RStudio, create a new R script and begin developing it by going to \sphinxstylestrong{File\textgreater{}New File\textgreater{}R Script}. In order to open the genome \sphinxcode{\sphinxupquote{vcf.gz}} files, \sphinxcode{\sphinxupquote{unzip}} and \sphinxcode{\sphinxupquote{tar}} are required. The system might have multiple directories containing \sphinxcode{\sphinxupquote{unzip}} and \sphinxcode{\sphinxupquote{tar}}.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{getOption}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{unzip}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{options}\PYG{p}{(}\PYG{n}{unzip} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{/usr/bin/unzip}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}}
\PYG{o}{\PYGZgt{}} \PYG{n}{Sys}\PYG{o}{.}\PYG{n}{getenv}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{TAR}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{Sys}\PYG{o}{.}\PYG{n}{setenv}\PYG{p}{(}\PYG{n}{TAR} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{/bin/tar}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

First install and load \sphinxcode{\sphinxupquote{devtools}}, which will help install the corresponding dependencies and GitHub libraries later on,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} install.packages(“devtools”)
\PYGZgt{} library(devtools)
\end{sphinxVerbatim}

Install the necessary dependencies using \sphinxhref{https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html}{BiocManager},

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{BiocManager}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install}\PYG{p}{(}\PYG{n}{c}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{VariantAnnotation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

If the above did not run, it may be needed to install first \sphinxhref{https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html}{BiocManager} and then run the above again,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{install}\PYG{o}{.}\PYG{n}{packages}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{BiocManager}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

Install the libraries \sphinxhref{https://github.com/knausb/vcfR}{vcfR} and \sphinxhref{https://github.com/AnzaGhaffar/GT-Plotting}{GT\sphinxhyphen{}Plotting} from their respective GitHub repositories,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{devtools}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install\PYGZus{}github}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{knausb/vcfR}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{devtools}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install\PYGZus{}github}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{AnzaGhaffar/GT\PYGZhy{}Plotting}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

Load the installed dependencies and GitHub libraries,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{library}\PYG{p}{(}\PYG{n}{tidyr}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{library}\PYG{p}{(}\PYG{n}{ggplot2}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{library}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{VariantAnnotation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{library}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{vcfR}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{library}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{GTPlotting}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}


\part{Pointing towards the vcf File}
\label{\detokenize{index:pointing-towards-the-vcf-file}}
RStudio will need to know where the \sphinxstylestrong{vcf} file is. Specify the path and filename,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{path} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{/path/to/directory/with/vcf/file/}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{o}{\PYGZgt{}} \PYG{n}{setwd}\PYG{p}{(}\PYG{n}{path}\PYG{p}{)}
\PYG{o}{\PYGZgt{}} \PYG{n}{vcffilename} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{name\PYGZus{}of\PYGZus{}vcf\PYGZus{}file.vcf.gz}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}

The next three steps consist of running functions to plot the genotypes.


\part{VcfToTable Function}
\label{\detokenize{index:vcftotable-function}}
\sphinxcode{\sphinxupquote{VcfToTable}} takes as input a \sphinxstylestrong{vcf} file with extension \sphinxcode{\sphinxupquote{.vcf}} or \sphinxcode{\sphinxupquote{.vcf.gz}} and creates an object that consists of two data frames,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZgt{}} \PYG{n}{vcf\PYGZus{}testdata}\PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}}\PYG{n}{VcfToTable}\PYG{p}{(}\PYG{n}{vcffilename}\PYG{p}{)}
\end{sphinxVerbatim}

Then, the important features are extracted from the \sphinxstylestrong{vcf} file for the genotype plotting using the \sphinxcode{\sphinxupquote{vcfdata}} data frame,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} vcf\PYGZus{}testdata\PYGZdl{}vcfdata
\end{sphinxVerbatim}

Running the above will output the CHROM, POS, REF, ALT, QUAL, INDVL1, and INDVL2. A data frame is created by running,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} vcf\PYGZus{}testdata\PYGZdl{}chromelen
\end{sphinxVerbatim}

The output looks like below with chromosome number and size,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
    \PYG{n}{chromosome}   \PYG{n}{size}
\PYG{l+m+mi}{1}  \PYG{n}{NC\PYGZus{}018051}\PYG{o}{.}\PYG{l+m+mi}{1}     \PYG{l+m+mi}{16}
\PYG{l+m+mi}{2}  \PYG{n}{NC\PYGZus{}040279}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{38450}
\PYG{l+m+mi}{3}  \PYG{n}{NC\PYGZus{}040280}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{34390}
\PYG{l+m+mi}{4}  \PYG{n}{NC\PYGZus{}040281}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{54830}
\PYG{l+m+mi}{5}  \PYG{n}{NC\PYGZus{}040282}\PYG{o}{.}\PYG{l+m+mi}{1} \PYG{l+m+mi}{193987}
\PYG{l+m+mi}{6}  \PYG{n}{NC\PYGZus{}040283}\PYG{o}{.}\PYG{l+m+mi}{1} \PYG{l+m+mi}{125079}
\PYG{l+m+mi}{7}  \PYG{n}{NC\PYGZus{}040284}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{36664}
\PYG{l+m+mi}{8}  \PYG{n}{NC\PYGZus{}040285}\PYG{o}{.}\PYG{l+m+mi}{1} \PYG{l+m+mi}{104691}
\PYG{l+m+mi}{9}  \PYG{n}{NC\PYGZus{}040286}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{58685}
\PYG{l+m+mi}{10} \PYG{n}{NC\PYGZus{}040287}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{83639}
\PYG{l+m+mi}{11} \PYG{n}{NC\PYGZus{}040288}\PYG{o}{.}\PYG{l+m+mi}{1} \PYG{l+m+mi}{276550}
\PYG{l+m+mi}{12} \PYG{n}{NC\PYGZus{}040289}\PYG{o}{.}\PYG{l+m+mi}{1}  \PYG{l+m+mi}{52588}
\end{sphinxVerbatim}


\part{GTPlotting\_Chromosome Function}
\label{\detokenize{index:gtplotting-chromosome-function}}
This function plots the genotype of each chromosome. It takes three inputs the \sphinxstylestrong{vcf} data frame generated by the \sphinxcode{\sphinxupquote{VcfToTable}} function, the chromosome length table generated by the \sphinxcode{\sphinxupquote{VcfToTable}} function, and the name of the control sample should be same as in the \sphinxstylestrong{vcf} file,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} GTPlotting\PYGZus{}Chromosome(vcf\PYGZus{}testdata\PYGZdl{}vcfdata,vcf\PYGZus{}testdata\PYGZdl{}chromelen,\PYGZsq{}Grinkan\PYGZus{}CTRL\PYGZsq{})
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=600\sphinxpxdimen]{{function_02_output}.jpeg}
\caption{“GTPlotting\_Chromosome” output. (click to expand)}\label{\detokenize{index:id2}}\end{figure}


\part{GTPlotting\_Chromosome\_Combined Function}
\label{\detokenize{index:gtplotting-chromosome-combined-function}}
This function plots the genotype of all the chromosomes. It takes two inputs the \sphinxstylestrong{vcf} data frame generated by the \sphinxcode{\sphinxupquote{VcfToTable}} function and the chromosome length table generated by the \sphinxcode{\sphinxupquote{VcfToTable}} function,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} GTPlotting\PYGZus{}Chromosome\PYGZus{}Combined(vcf\PYGZus{}testdata\PYGZdl{}vcfdata,vcf\PYGZus{}testdata\PYGZdl{}chromelen)
\end{sphinxVerbatim}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=600\sphinxpxdimen]{{function_03_output}.jpeg}
\caption{GTPlotting\_Chromosome\_Combined” output. (click to expand)}\label{\detokenize{index:id3}}\end{figure}



\renewcommand{\indexname}{Index}
\printindex
\end{document}